src/drivers/cmakeFileApiDriver.ts

```diff
@@ -257,14 +256,20 @@ export class CMakeFileApiDriver extends CMakeDriver {
             log.info(proc.buildCmdStr(this.cmake.path, args));
             return 0;
         } else {
-            log.debug(`Configuring using ${this.useCMakePresets ? 'preset' : 'kit'}`);
+            if (!taskCustomConfig) {
+                log.debug(`Configuring using ${this.useCMakePresets ? 'preset' : 'kit'}`);
+            } else {
+                log.debug("Configuring using tasks");
+            }
             log.debug('Invoking CMake', cmake, 'with arguments', JSON.stringify(args));
             const env = await this.getConfigureEnvironment();
             const res = await this.executeCommand(cmake, args, outputConsumer, { environment: env, cwd: this.binaryDir }).result;
             log.trace(res.stderr);
             log.trace(res.stdout);
             if (res.retc === 0) {
-                this._needsReconfigure = false;
+                if (!taskCustomConfig) {
```

Comment:
a task can configure using a different config preset than the default config preset. 

If it is not a task related configure, it should be set to false,
if it is a task, it can be set to false if the config preset is the same as the default preset. we don't have the preset name info in this function, we can send it though.